### DISCUSSION TITLE
specific car model scan doesn't seem to work if not a trigram

### DISCUSSION THREAD

**Discussion by cleb40** [cedricp/ddt4all#1289] [Q&A]
Hello,
car : renault megane 2 phase 2 ( french ) , OS : debian 12 , python : 3.12.2 , OBD2 : vgate vlinker MC+, ddt4all : 3.0.2 ( git clone )

i already noticed that before in demo mode , but if i filter the ECU i want to scan by car model  ( like X84 or X84Ph2 ) the scan only lauch for trigrams , per say : X84 will work while X84Ph2 not, i tried different models in demo mode, and it was the same (dummy data only shows with trigrams ).
Though i noticed , my car ecus ( some ) shows up with a X84 scan because they can be included in multiples project as i could verify in the ecu/db.json

Is this the intended behavior ?

On the side since i found an OBD2 interface which seems to work ( so far ) i'd like to share the info here on github , but there doesn't seem to be any page acting as a database of compatible interfaces ( including  where it was bought at what price and the result of elm benchmark for exemple )
Would be nice to do that... ( and maybe would have saved me a lot of time against  chinese crap )

**Comment by cleb40** [comment]
Hello,
after a bit of hacking , i found out that it is due to these line in ecu.py :  projname = proj[0:3].upper()
used when parsing database either ecu/*json or ecu.zip or xml files , so it appears 3 times in the code.
as of now appears at lines : 

for json files : 1238
for ecu.zip    :  1282
for xml files   :  1344

if i change it with for exemple projname = proj[0:18].upper()
the dummy scan works

though i don't know the impact, so i will wait for an answer here before trying with my car.

**Comment by cleb40** [comment]
transformed into an issue, closing

**Comment by Furtif** [comment]
This in general does not talk about vgate in readme and/or any other wiki, we do not take your accessories into account. If they work better, but that is no reason to issue issues on material that has never been mentioned and/or tested by the development team.

If you find an error with an extra accessory and are able to resolve it, you can propose this in readme along with the PR (pull request) that allows it.

Cordially
